doc/Language/js-nutshell.pod6

```diff
@@ -393,9 +393,100 @@ say ++$foo; # 0
 
 =head1 The networking API
 
-=comment TODO
+=head2 Net
 
-# TBD
+In Perl 6, there are two APIs for dealing with networking: C<IO::Socket::INET>
+(for synchronous networking), and  C<IO::Socket::Async> (for asynchronous
+networking).
+
+C<IO::Socket::INET> currently only supports TCP connections. Its API resembles
+that of C's socket API. If you're familiar with that, then it won't take long
+to understand how to use it. For example, here's an example of an echo server
+that closes after receiving one message:
+
+=begin code
+my IO::Socket::INET $server .= new:
+    :localhost<localhost>,
+    :localport<8000>,
+    :listen;
+
+my IO::Socket::INET $client .= new: :host<localhost>, :port<8000>;
+$client.print: 'ayy lmao';
+
+my IO::Socket::INET $conn = $server.accept;
+my Str $msg               = $conn.recv;
+$conn.write($msg);
+
+say $client.recv.decode; # RESULT: ayy lmao
+$conn.close;
+$client.close;
+$server.close;
+=end code
+
+By default, C<IO::Socket::INET> connections are IPv4 only. To use IPv6 instead,
+pass C<:family(PF_INET6)> when constructing a server or a client.
+
+In contrast, C<IO::Socket::Async> supports both IPv4 and IPv6 without the need
+to specify which family you wish to use. It also supports UDP sockets. Here's
+how you would write the same echo server as above asynchronously (note that
+C<Supply.tap> is multithreaded; if this is undesirable, use C<Supply.act>
+instead:
+
+=begin code
+my Promise $p .= new;
+my Vow     $v  = $p.vow;
+
+my $server = IO::Socket::Async.listen('localhost', 8000);
+my $tap    = $server.tap(-> $conn {
+    $conn.Supply.tap(-> $data {
+        $conn.write: $data;
+        $v.keep: True;
+    })
+});
+
+my $client = IO::Socket::Async.connect('localhost', 8000);
+$client.print: 'ayy lmao';
+await $p;
+$client.close;
+$server.close;
+=end code
+
+The equivalent code in Node.js looks like this:
+
+=begin code :lang<javascript>
+const net = require('net');
+
+const server = net.createServer(conn => {
+    conn.on('data', data => {
+        conn.write('ayy lmao');
+    });
+}).listen(8000, 'localhost');
+
+const client = net.createConnection(8000, 'localhost', () => {
+    client.on('data', () => {
+        client.close;
+        server.close;
+    });
+
+    client.write("ayy lmao");
+});
+=end code
+
+=head2 HTTP/HTTPS
+
+Perl 6 doesn't natively support HTTP/HTTPS. However, CPAN packages such as
+L<Cro|https://cro.services/> help fill the gap.
+
+=head2 DNS
+
+Perl 6 does not support DNS currently, though there are some CPAN modules in
```

Comment:
This makes it read as if I can't use a domain name as argument to `connect`. I don't use the sockets myself, so I'm not sure if this is actually true, but generally DNS lookups are handled by the OS and should Just Work. Is it indeed not possible to use something like `IO::Socket::Async.connect('tyil.nl', 8000);`?

Reply:
It is possible. What I was trying to say is that most of the features of Node's DNS module, like record lookups and reverse IP lookups, aren't yet implemented. I'll reword it